gems/pending/spec/appliance_console/database_replication_spec.rb

```diff
@@ -0,0 +1,167 @@
+require "appliance_console/logging"
+require "appliance_console/prompts"
+require "appliance_console/database_replication"
+require "appliance_console/logging"
+require "tempfile"
+
+describe ApplianceConsole::DatabaseReplication do
+  before do
+    allow(subject).to receive(:say)
+    allow(subject).to receive(:clear_screen)
+    allow(subject).to receive(:agree)
+    allow(subject).to receive(:ask_for_ip_or_hostname)
+    allow(subject).to receive(:ask_for_password_or_none)
+    allow(subject).to receive(:ask_for_password)
+  end
+
+  context "#ask_for_unique_cluster_node_number" do
+    it "should ask for a unique number" do
+      expect(subject).to receive(:ask_for_integer).with(/uniquely identifying this node/i).and_return(1)
+
+      subject.ask_for_unique_cluster_node_number
+    end
+  end
+
+  context "#ask_for_database_credentials" do
```

Comment:
I'm not sure what the distinction between the three tests here is. It feels like we are testing the behavior of `ask_for_password` rather than making sure the database credentials are actually set on the subject object (which is the real purpose of these methods).

i.e. I would expect something like:
`expect(subject.database_name).to eq("new_database_name")`
to show up somewhere here.